pkg/addon/utils.go

```diff
@@ -279,23 +277,58 @@ func IsAddonDir(dirName string) (bool, error) {
 		return false, errors.Errorf("addon version is empty")
 	}
 
-	// Load template.yaml
-	templateYaml := filepath.Join(dirName, TemplateFileName)
-	if _, err := os.Stat(templateYaml); os.IsNotExist(err) {
-		return false, errors.Errorf("no %s exists in directory %q", TemplateFileName, dirName)
+	// Load template.yaml/cue
+	var errYAML error
+	var errCUE error
+	templateYAML := filepath.Join(dirName, TemplateFileName)
+	templateCUE := filepath.Join(dirName, AppTemplateCueFileName)
+	_, errYAML = os.Stat(templateYAML)
+	_, errCUE = os.Stat(templateCUE)
+	if os.IsNotExist(errYAML) && os.IsNotExist(errCUE) {
+		return false, fmt.Errorf("no %s or %s exists in directory %q", TemplateFileName, AppTemplateCueFileName, dirName)
+	}
+	if errYAML != nil && errCUE != nil {
+		return false, errors.Errorf("cannot stat %s or %s", TemplateFileName, AppTemplateCueFileName)
 	}
-	templateYamlContent, err := ioutil.ReadFile(filepath.Clean(templateYaml))
+
+	// template.cue have higher priority
+	if errCUE == nil {
+		templateContent, err := ioutil.ReadFile(filepath.Clean(templateCUE))
+		if err != nil {
+			return false, fmt.Errorf("cannot read %s: %w", AppTemplateCueFileName, err)
+		}
+		v, err := value.NewValue("", nil, "")
+		if err != nil {
+			return false, err
+		}
+		out, err := v.LookupByScript(string(templateContent))
+		if err != nil {
+			return false, err
+		}
+		content, err := out.LookupValue(renderOutputCuePath)
+		if err != nil {
+			return false, fmt.Errorf("no output in %s: %w", AppTemplateCueFileName, err)
+		}
+		app := &v1beta1.Application{}
+		err = content.UnmarshalTo(app)
+		if err != nil {
+			return false, fmt.Errorf("no application in %s: %w", AppTemplateCueFileName, err)
+		}
+		return true, nil
+	}
```

Comment:
please refer to https://github.com/kubevela/kubevela/pull/4443/files

Reply:
I figured out that if I were to Unmarshal to Application, it required all these files, otherwise it won't work. Since it only determines if it is a addon dir, just look for `output` key word is enough.

Is this enough?